r/geoarrow/R/sf-compat.R

```diff
@@ -5,6 +5,37 @@ infer_nanoarrow_schema.sfc <- function(x, ...) {
   infer_geoarrow_schema(x)
 }
 
-# Eventually we can add specializations for as_geoarrow_array() based on
-# st_as_grob(), which is very fast and generates lengths + a column-major
-# matrix full of buffers we can provide views into.
+#' @export
+as_geoarrow_array.sfc <- function(x, ..., schema = NULL) {
+  # Let the default method handle custom output schemas
+  if (!is.null(schema)) {
+    return(NextMethod())
+  }
+
+  # Let the default method handle M values (the optimized path doesn't
+  # handle mixed XYZ/XYZM/XYM but can deal with mixed XY and XYZ)
+  if (!is.null(attr(x, "m_range"))) {
+    return(NextMethod())
+  }
+
+  if (class(x)[1] %in% c("sfc_POINT",
```

Comment:
Maybe use wk::wk_vector_meta() here?
```suggestion
  # or use the labels instead
  if (wk::wk_vector_meta(x)$geometry_type %in% 1:6) {
```